src/app/teacher/abstract-tag-dialog.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Directive, OnInit } from '@angular/core';
+import {
+  FormControl,
+  Validators,
+  ValidatorFn,
+  AbstractControl,
+  ValidationErrors
+} from '@angular/forms';
+import { ProjectTagService } from '../../assets/wise5/services/projectTagService';
+import { Tag } from '../domain/tag';
+
+@Directive()
+export abstract class AbstractTagDialogComponent implements OnInit {
+  protected tagControl = new FormControl('', [
+    Validators.required,
+    this.createUniqueTagValidator()
+  ]);
+  protected tags: Tag[] = [];
```

Comment:
If you move the order of these variables, you won't need to check for null in ```createUniqueTagValidator()``` 
```suggestion
    protected tags: Tag[] = [];
    protected tagControl = new FormControl('', [
    Validators.required,
    this.createUniqueTagValidator()
  ]);
```